File: aexample/src/main/java/org/judal/examples/java/jdbc/E05_ExportTablesToJDOXML.java
package org.judal.examples.java.jdbc;

import org.junit.Test;

import org.judal.metadata.SchemaMetaData;
import org.judal.metadata.bind.JdoXmlMetadata;

import org.judal.jdbc.JDBCRelationalDataSource;

import org.judal.examples.StudentCourseSchema;

/**
 * Programmatically create schema meta data and output it as JDO XML
 */
public class E05_ExportTablesToJDOXML {

	@Test
	public void demo() throws Exception {

		JDBCRelationalDataSource dataSource = E02_CreateAdditionalDataSource.create();
		
		SchemaMetaData schemaMetaData = StudentCourseSchema.generateSchemaMetaData(dataSource);
		
		JdoXmlMetadata xmlMeta = new JdoXmlMetadata (dataSource);

		xmlMeta.writeMetadata(schemaMetaData, System.out);

		E02_CreateAdditionalDataSource.close(dataSource);
	}

}

File: aexample/src/main/scala/org/judal/examples/scala/jdbc/E05_ExportTablesToJDOXML.scala
package org.judal.examples.scala.jdbc

import org.junit.Test

import org.scalatest.Suite

import org.judal.metadata.bind.JdoXmlMetadata

import org.judal.examples.StudentCourseSchema

/**
 * Example of how to programmatically create schema meta data and output it as JDO XML
 */
class E05_ExportTablesToJDOXML extends Suite {

	@Test def demo() = {

		val dataSource = E02_CreateAdditionalDataSource.create()
		
		val schemaMetaData = StudentCourseSchema.generateSchemaMetaData(dataSource)
		
		val xmlMeta = new JdoXmlMetadata (dataSource)

		xmlMeta.writeMetadata(schemaMetaData, System.out)

		E02_CreateAdditionalDataSource.close(dataSource)
	}

}
